cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -266,6 +266,7 @@ public void create(Id.Stream streamId) throws Exception {
 
   @Override
   public void create(final Id.Stream streamId, @Nullable final Properties props) throws Exception {
+    assertNamespaceHomeExists(streamId.getNamespace());
```

Comment:
No, if we have to check, its fine here. Just wondering if the check is needed at all, since the Handler ensures that the namespace exits.